plugin/trino-exchange-filesystem/src/main/java/io/trino/plugin/exchange/filesystem/s3/S3FileSystemExchangeStorage.java

```diff
@@ -144,6 +169,24 @@ public S3FileSystemExchangeStorage(S3FileSystemExchangeStorageStats stats, Excha
                 config.getAsyncClientConcurrency(),
                 config.getAsyncClientMaxPendingConnectionAcquires(),
                 config.getConnectionAcquisitionTimeout());
+
+        if (compatibilityMode == GCP) {
+            if (config.getGcsJsonKeyFilePath().isPresent()) {
+                Credentials credentials = GoogleCredentials.fromStream(new FileInputStream(config.getGcsJsonKeyFilePath().get()));
+                this.gcsClient = Optional.of(StorageOptions.newBuilder().setCredentials(credentials).build().getService());
+            }
+            else {
+                this.gcsClient = Optional.of(StorageOptions.getDefaultInstance().getService());
+            }
+            this.gcsDeleteExecutor = Optional.of(MoreExecutors.listeningDecorator(new ThreadPoolExecutor(0, 100,
```

Comment:
nit: one parameter per line, static import `listeningDecorator`

Reply:
Also set the core pool size to 100, otherwise it will keep running only a single thread until the queue is full

Reply:
Ah you are right. I'm using a `SynchronousQueue` here, which basically has a size of 0, and if concurrent tasks exceed 100, rejection will happen. Your suggestion above is better.